src/command_modules/azure-cli-network/azure/cli/command_modules/network/tests/test_network_commands.py

```diff
@@ -574,6 +574,36 @@ def body(self):
         s.cmd('network public-ip list -g {}'.format(rg), checks=JMESPathCheck("length[?name == '{}']".format(public_ip_dns), None))
 
 
+@api_version_constraint(ResourceType.MGMT_NETWORK, min_api='2017-06-01')
+class NetworkRouteFilterScenarioTest(ScenarioTest):
+
+    @ResourceGroupPreparer(name_prefix='cli_test_network_route_filter')
+    def test_network_route_filter(self, resource_group):
+        kwargs = {
+            'rg': resource_group,
+            'filter': 'filter1'
+        }
+        self.cmd('network route-filter create -g {rg} -n {filter}'.format(**kwargs))
+        self.cmd('network route-filter update -g {rg} -n {filter}'.format(**kwargs))
+        self.cmd('network route-filter show -g {rg} -n {filter}'.format(**kwargs))
+        self.cmd('network route-filter list -g {rg}'.format(**kwargs))
+
+        self.cmd('network route-filter rule list-service-communities')
+        try:
+            self.cmd('network route-filter rule create -g {rg} --filter-name {filter} -n rule1 --communities 12076:5040 12076:5030 --access allow'.format(**kwargs))
+        except CLIError:
+            pass
```

Comment:
Is this testing expected failure? If so should assert fail if the exception is *not* thrown?

Reply:
I will try re-recording now.  It was strange because the command would fail saying the feature wasn't enabled... yet it would actually create the rule.  🤷‍